src/RCONServer.h

```diff
@@ -35,72 +33,61 @@ class cRCONServer :
 	
 protected:
 	friend class cRCONCommandOutput;
+	friend class cRCONListenCallbacks;
 	
 	class cConnection :
-		public cSocketThreads::cCallback
+		public cTCPLink::cCallbacks
 	{
 	public:
-		cConnection(cRCONServer & a_RCONServer, cSocket & a_Socket);
+		cConnection(cRCONServer & a_RCONServer, const AString & a_IPAddress);
 		
 	protected:
 		friend class cRCONCommandOutput;
 	
-		/// Set to true if the client has successfully authenticated
+		/** Set to true if the client has successfully authenticated */
 		bool m_IsAuthenticated;
 		
-		/// Buffer for the incoming data
+		/** Buffer for the incoming data */
 		AString m_Buffer;
 		
-		/// Buffer for the outgoing data
-		AString m_Outgoing;
-		
-		/// Server that owns this connection and processes requests
+		/** Server that owns this connection and processes requests */
 		cRCONServer & m_RCONServer;
 		
-		/// The socket belonging to the client
-		cSocket & m_Socket;
+		/** The TCP link to the client */
+		cTCPLinkPtr m_Link;
 		
-		/// Address of the client
+		/** Address of the client */
 		AString m_IPAddress;
 		
 
-		// cSocketThreads::cCallback overrides:
-		virtual bool DataReceived(const char * a_Data, size_t a_Size) override;
-		virtual void GetOutgoingData(AString & a_Data) override;
-		virtual void SocketClosed(void) override;
+		// cTCPLink::cCallbacks overrides:
+		virtual void OnLinkCreated(cTCPLinkPtr a_Link);
+		virtual void OnReceivedData(const char * a_Data, size_t a_Length) override;
+		virtual void OnRemoteClosed(void) override;
+		virtual void OnError(int a_ErrorCode, const AString & a_ErrorMsg) override;
 		
-		/// Processes the given packet and sends the response; returns true if successful, false if the connection is to be dropped
-		bool ProcessPacket(int a_RequestID, int a_PacketType, int a_PayloadLength, const char * a_Payload);
+		/** Processes the given packet and sends the response; returns true if successful, false if the connection is to be dropped */
+		bool ProcessPacket(UInt32 a_RequestID, UInt32 a_PacketType, UInt32 a_PayloadLength, const char * a_Payload);
 		
-		/// Reads 4 bytes from a_Buffer and returns the int they represent
-		int IntFromBuffer(const char * a_Buffer);
+		/** Reads 4 bytes from a_Buffer and returns the LE UInt32 they represent */
```

Comment:
I don't think so. The value is represented LE in the buffer; storing the value in a specific endian in the `UInt32` is something no-one would really do.

But if you have a better wording, go ahead and change it by all means. I'm not a native English speaker, after all.